game-core/src/main/java/games/strategy/engine/framework/headlessGameServer/HeadlessGameServer.java

```diff
@@ -244,7 +247,7 @@ private static String hashPassword(final String password, final String salt) {
 
   public String remoteShutdown(final String hashedPassword, final String salt) {
     final String password = System.getProperty(LOBBY_GAME_SUPPORT_PASSWORD, "");
-    if (password.equals(GameRunner.NO_REMOTE_REQUESTS_ALLOWED)) {
+    if (password.equals(NO_REMOTE_REQUESTS_ALLOWED)) {
```

Comment:
General Information:
I just reverse engineered the evil witchcraft behind with weird password mechanism.
Turns out that the lobby gets a random salt from a bot, hashes the guessed password with that salt, sends it over the internet, where it's being compared a couple lines further down.
So it's basically a "encryption mechanism" with a challenge, with the difference, that usually the password should be the encryption key, not the challenge

Reply:
:+1: Good to know.  I'm sure I'll run into that again while trying to migrate code to `game-headless`.